src/app/userDashboard/store/effects.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { switchMap, map, retry, catchError, EMPTY } from 'rxjs';
+import { CrudService } from '../services/crud.service';
+import {
+  CoreActionsUnion,
+  getTripsFromFirestore,
+  addTripToFirestore,
+  getTripsComplete,
+  deleteTripFromFirestore,
+} from './actions';
+
+@Injectable()
+export class UserDashboardEffects {
+  addTripToFirestore$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(addTripToFirestore.type),
+      switchMap((action) =>
+        this.crudService
+          .addTrip(action.userEmail, action.tripName, action.destination)
+          .pipe(
+            map(() => {
+              return getTripsFromFirestore({
+                userEmail: action.userEmail,
+              });
+            }),
+            retry(1),
+            catchError((err) => {
+              alert(
+                `Unfortunately we could not add ${action.tripName} to the database. Please try adding it again.` +
```

Comment:
This is just a personal preference but maybe start the alert with the err.toString() then add a next line character ` \n ` (maybe two of them) then afterwards you can add the understandable error message

Reply:
I thought that since this error is for the user mostly, it would be better feedback if i lead with the more understandable error message. But i'll add the next line characters.